### ISSUE TITLE
Increasingly common 403 errors when casting videos

### ISSUE THREAD

**Issue by nickrbogdanov** [anxdpanic/plugin.video.youtube#501]
### Context

Please provide any relevant information about your setup

* Add-on Version: git master branch
* Kodi Version: Nexus
* Kodi GUI Language: English
* Operating System: Ubuntu Linux
* Operating System Language: English

Since about March, this plugin has worked great.  In the past few weeks, I've seen more and more instances where I send a video from Cast Kodi and a transient error occurs:

```
2023-08-07 21:37:36.347 T:921805   error <general>: CCurlFile::CReadState::FillBuffer - (0x7fbc7838bc20) Failed: HTTP returned code 403
2023-08-07 21:37:36.347 T:921805   error <general>: CCurlFile::Open - <https://rr1---sn-o097znsd.googlevideo.com/videoplayback?expire=1691490717&ei=PcXRZKO6IYz_kgaKoYMI&ip=2601%3A[REDACTED] Failed with code 403:
2023-08-07 21:37:36.347 T:921805   error <general>: CFileCache::Open - <https://rr1---sn-o097znsd.googlevideo.com/videoplayback?expire=1691490717&ei=[REDACTED]&ip=2601%3A[REDACTED]-> failed to open
2023-08-07 21:37:36.347 T:921805   error <general>: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.youtube/play/?video_id=EMy_QuFEUZw]
2023-08-07 21:37:36.347 T:921805    info <general>: CVideoPlayer::OnExit()
2023-08-07 21:37:36.353 T:880936    info <general>: CVideoPlayer::CloseFile()
2023-08-07 21:37:36.355 T:880936    info <general>: VideoPlayer: waiting for threads to exit
```

On the UI I see: "One or more items failed to play. Check the log for more information about this message."

Usually I retry a few times and it works again.  This is on a regular Comcast internet connection, no VPNs or anything weird.

Can we make the plugin automatically retry on its own so this annoyance goes away?

**Comment by MoojMidge** [comment]
I was never able to reliably replicate this, but I think it is already fixed.

Can you test using this [test version](https://github.com/MoojMidge/plugin.video.youtube/archive/refs/heads/master.zip) and see if this resolves the problem?

**Comment by nickrbogdanov** [comment]
Seems to be working so far, thanks.  Will run for a few days and update the ticket with my results.

**Comment by DjDiabolik** [comment]
Hi @MoojMidge last night on my actual Kodi 20.2 (running on Ubuntu 22.04.3 LTS) when try to reproduce some youtube video i obtain a strange trouble:
Whit totally random situation the video freeze (kodi not crash.. i can open OSD and stop the reproduction of video) sometime also very first 10 / 15 seconds and audio become in glitch like a infinite loop.
I can skip forward or backward but again after 10 / 15 seconds freeze again.........


Looking inside current kodi.log appears so many errors.. somethings like:
' inputstream.adaptive....Failed to download segment......... cause 403 etc etc.'

I have tryed to google about this specific errors and i have reach this issue............ after that i have installed your current 'TEST version' and apparently the same video now is played correctly whitout any issue.

So also on current kodi.log it's appears so many errors say:
"CAddonsSetting plugins.video.youtube Unable to read value"8" of 'kodion.mpd.quality.selection'" or something similar.

Do you have some idea how it's related ?

**Comment by MoojMidge** [comment]
It is not related. The test version has changed settings while you are still using old invalid settings. Open the Youtube plugins settings and press OK to save the new defaults for settings that have changed, or just change the new settings to whatever you want.

Edit: An invalid setting value shouldn't create "so many errors". The first time the error occurs the setting value should be set to the default. If you have a problem create a new issue and post a log. Trying to guess what is happening from the bits and pieces of information you are providing is a massive waste of time.

**Comment by DjDiabolik** [comment]
> It is not related. The test version has changed settings while you are still using old invalid settings. Open the Youtube plugins settings and press OK to save the new defaults for settings that have changed, or just change the new settings to whatever you want.
> 
> Edit: An invalid setting value shouldn't create "so many errors". The first time the error occurs the setting value should be set to the default. If you have a problem create a new issue and post a log. Trying to guess what is happening from the bits and pieces of information you are providing is a massive waste of time.

Lol i know sorry..... anyway your testing as helped a lot to resolve the issue whit freezing video reproduction.
Speaking about freeze video if it's append again on my setup i try to provide a new kodi.log whit debug log enabled.

Anyway.... i have open the setting on youtube addons. Navigate on some settings and applyed some desidered option and saved it. Apparently all voices about 'unable read value etc etc" it's disappears :)

**Comment by MoojMidge** [comment]
All good, just remember that if you need help, don't make it unnecessarily difficult for people to help you.

**Comment by MoojMidge** [comment]
Closed by #506
